### ISSUE TITLE
[Bug Report] Naming issue for Taiwan in list of countries

### ISSUE THREAD

**Issue by null-von-sushi** [stashapp/stash#5237]
**Describe the bug**
Taiwan is currently labelled a "province of China". This name is only used by communist China and countries/institutions that pander to them, and does not represent the people of Taiwan. 

Officially the island is a province of the *Republic of China*, however the current phrasing makes it seems as if it was subservient to the *People's Republic of China* (the communist one). Similar situation like north and south korea going on, basically.

**To Reproduce**
1. Go to performers, and edit any performer. 
2. Click on 'Country'
3. Scroll down to 'Taiwan'
4. See issue.

**Expected behavior**
The name should just be `Taiwan` (ideal and neutral) or `Republic of China (Taiwan)` (the official phrasing used by the government but not as neutral). 

**Screenshots**
![image](https://github.com/user-attachments/assets/bb5659ce-c496-43d2-954e-f1a046c93a21)

**Stash Version: (from Settings -> About):**
` ` `
v0.26.2
Build hash:
2a373a25
Build time:
2024-06-27 00:28:19
` ` `

**Desktop (please complete the following information):**
 - OS: Arch Linux
 - Browser: Firefox

**Additional context**
Set it up today, running via docker compose via CasaOS on my server.

**Comment by WithoutPants** [comment]
We get the country list from `i18n-iso-countries`, which in turn is from the ISO codes: https://www.iso.org/obp/ui/#iso:code:3166:TW
 
See also: https://github.com/michaelwittig/node-i18n-iso-countries/issues/266

I'm not overly inclined to make local code changes to override this behaviour in the stash code that implies a given political stance.

**Comment by WithoutPants** [comment]
Having considered this further, I'll look into adding some functionality for users to be able to customise this on their local systems.

**Comment by null-von-sushi** [comment]
Thank you very much 🙏

**Comment by AutisticShark** [comment]
@null-von-sushi I noticed your issue on immich-app/immich. I dug deeper yesterday, and it turns out to be a supply chain attack by a Chinese user. The offending PR was opened around 2021, and the repo owner of `i18n-iso-countries` merged that PR without question, and he refused any future revert effort from other users to be "neutral". I suggest we replace the `i18n-iso-countries` with `i18n-iso-3166-1`, which is a package created by me to revert that particular PR and without changing any API interface of the package.

**Comment by null-von-sushi** [comment]
While "country/region" might make the label more flexible, the issue is about taiwan's incorrect naming. Taiwan is not "universally recognized" as a province of China (not even de jure as there's countries that do have official diplomatic relations), and de facto it is a sovereign country that has never been ruled by the communist party of China, has its own passport, currency, military etc. in fact, the taiwanese passport has more visa free access to more countries than the Chinese one. 

tl;Dr:
• Country -> "Country/Region" would be an improvement as it might better describ some edge cases
• Taiwan being a "Province of China" is misleading, and the most recent comment is just an attempt to establish the PRC viewpoint as common.

**Comment by null-von-sushi** [comment]
Imma be honest, this was hard to decipher and reads more like either an ad hominem attack or a schizophrenic episode...
Regarding point 1, I never disagreed that region would make the software worse or more ambiguous? But that's not related to this issue, it's offtopic. Regarding 2, it's clear you're pushing this to try and further spread CCP rhetoric as fact.
> which is the consensus of the vast majority of countries in the international community

But not all countries, but that's beside the point because even if group of people were to come together and declare 'the sky is red!' that would not make it so. Anyone who has even a little more than surface knowledge of the matter can see that.

> (I'm sure you're not one of them, lol!)

I am not sure what you're talking about. My country of residence or nationality? I'm European dude. Greeting from the rainy UK.

> , and there is no dissent

???

> they should persuade his "LAI pseudo-regime" to quickly establish a nation.

I am genuinely not sure what to respond to this, except 'please take your medication' and maybe consider stop violating your countries law and stay on the side of the firewall that aligns with your delusions.

----

This will be my last post here unless either the developer responds or the conversation returns to something more technical. I don't want this to be burried in a flame war.


For anyone finding this via search or in the future, please do not be mislead by people online. Information is available (and not even hidden, literally just wikipedia or a youtube video on history or something. Or if you don't believe what's written on the internet, just go to a People's Republic of China embassy and try to get a visa for Taiwan, see what happens).

**Comment by DogmaDragon** [comment]
As mentioned by @WithoutPants, we are open to adding functionality to allow users to configure it on their own systems locally (PRs welcome). In the meantime we are planning on continuing using [node-i18n-iso-countries](https://github.com/michaelwittig/node-i18n-iso-countries) library.

You are welcome to suggest upstream changes there or to https://iso.org.
